profiles/identity/host-identity-by-vault.nix

```diff
@@ -4,8 +4,22 @@
 
   datacenter = config.currentCoreNode.datacenter or config.cluster.region;
 
-  reload = service: "${pkgs.systemd}/bin/systemctl try-reload-or-restart ${service}";
-  restart = service: "${pkgs.systemd}/bin/systemctl try-restart ${service}";
+  agentCommand = pathPkgs: namePrefix: cmds: let
+    script = pkgs.writeShellScriptBin "${namePrefix}.sh" ''
```

Comment:
Use `writeShellApplication` here. (also needs only `set -x` and `PATH` setting can be replaced with `runtimeInputs` then)